test/ILLink.RoslynAnalyzer.Tests/UnconditionalSuppressMessageCodeFixTests.cs

```diff
@@ -0,0 +1,512 @@
+using Xunit;
+using VerifyCSUSMwithRAF = ILLink.RoslynAnalyzer.Tests.CSharpCodeFixVerifier<
+	ILLink.RoslynAnalyzer.RequiresAssemblyFilesAnalyzer,
+	ILLink.CodeFix.UnconditionalSuppressMessageCodeFixProvider>;
+using VerifyCSUSMwithRUC = ILLink.RoslynAnalyzer.Tests.CSharpCodeFixVerifier<
+	ILLink.RoslynAnalyzer.RequiresUnreferencedCodeAnalyzer,
+	ILLink.CodeFix.UnconditionalSuppressMessageCodeFixProvider>;
+
+namespace ILLink.RoslynAnalyzer.Tests
+{
+	public class UnconditionalSuppressMessageCodeFixTests
+	{
+		private const string rucDef = @"
+#nullable enable
+namespace System.Diagnostics.CodeAnalysis
+{
+	[AttributeUsage(AttributeTargets.Constructor | AttributeTargets.Method, Inherited = false)]
+	public sealed class RequiresUnreferencedCodeAttribute : Attribute
+	{
+		public RequiresUnreferencedCodeAttribute(string message) { Message = message; }
+		public string Message { get; }
+		public string? Url { get; set; }
+	}
+}";
+		private const string rafDef = @"
+#nullable enable
+namespace System.Diagnostics.CodeAnalysis
+{
+	[AttributeUsage(AttributeTargets.Constructor | AttributeTargets.Event | AttributeTargets.Method | AttributeTargets.Property, Inherited = false, AllowMultiple = false)]
+	public sealed class RequiresAssemblyFilesAttribute : Attribute
+	{
+		public RequiresAssemblyFilesAttribute () { }
+		public string? Message { get; set; }
+		public string? Url { get; set; }
+	}
+}";
+		private const string usmDef = @"
+namespace System.Diagnostics.CodeAnalysis
+{
+	[AttributeUsage(AttributeTargets.All, Inherited = false, AllowMultiple = true)]
+    public sealed class UnconditionalSuppressMessageAttribute : Attribute
+    {
+        public UnconditionalSuppressMessageAttribute (string category, string checkId)
+		{
+			Category = category;
+			CheckId = checkId;
+		}
+		public string Category { get; }
+		public string CheckId { get; }
+		public string? Scope { get; set; }
+		public string? Target { get; set; }
+		public string? MessageId { get; set; }
+		public string? Justification { get; set; }
+	}
+}";
+
+
+		static Task VerifyUnconditionalSuppressMessageCodeFixWithRUC (
+			string source,
+			string fixedSource,
+			DiagnosticResult[] baselineExpected,
+			DiagnosticResult[] fixedExpected,
+			int? numberOfIterations = null)
+		{
+			var test = new VerifyCSUSMwithRUC.Test {
+				TestCode = source + rucDef + usmDef,
+				FixedCode = fixedSource + rucDef + usmDef,
+			};
+			test.ExpectedDiagnostics.AddRange (baselineExpected);
+			test.TestState.AnalyzerConfigFiles.Add (
+						("/.editorconfig", SourceText.From (@$"
+is_global = true
+build_property.{MSBuildPropertyOptionNames.EnableTrimAnalyzer} = true")));
+			if (numberOfIterations != null) {
+				test.NumberOfIncrementalIterations = numberOfIterations;
+				test.NumberOfFixAllIterations = numberOfIterations;
+			}
+			test.FixedState.ExpectedDiagnostics.AddRange (fixedExpected);
+			return test.RunAsync ();
+		}
+
+		static Task VerifyUnconditionalSuppressMessageCodeFixWithRAF (
```

Comment:
Is the above method identical except for the name of the MSBuild property name? If so, consider extracting a helper.

Reply:
The type each function is dealing with, so I  create a generic helpers